portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,50 +32,56 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> storeCommentList = new ArrayList<String>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    String json = convertToJsonUsingGson(storeCommentList);
+    Query query = new Query("UserComment").addSort("Timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-    response.setContentType("application/json");
-    response.getWriter().println(json);
-  }
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> userCommentList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = (long) entity.getKey().getId();
+      String firstName = (String) entity.getProperty("FirstName");
+      String lastName = (String) entity.getProperty("LastName");
+      String subjectText = (String) entity.getProperty("Subject");
+      String messageDescription = (String) entity.getProperty("Message");
+      long timestamp = (long) entity.getProperty("Timestamp");
+
+      String myUserComment = "Name: " + firstName + " " + lastName + "\n" + "ID: " + id + "\n Subject: " + subjectText + "\n" + messageDescription + "\n" + id;
+      userCommentList.add(myUserComment);
+    }
 
-  private String convertToJsonUsingGson(ArrayList storeCommentList) {
     Gson gson = new Gson();
-    String json = gson.toJson(storeCommentList);
-    return json;
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(userCommentList));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    // Get the input from the form.
+    // Records input from the contact form.
     String firstName = request.getParameter("firstname");
     String lastName = request.getParameter("lastname");
     String subjectText = request.getParameter("subject");
-    String messageDescritpion = request.getParameter("message");
+    String messageDescription = request.getParameter("message");
+    long timestamp = System.currentTimeMillis();
 
-    String myUserComment = "Name: " + firstName + " " + lastName + "\n" + "Subject: " + subjectText + "\n" + messageDescritpion;
-    storeCommentList.add(myUserComment);
+    // Initiate the Datastore service for storage of entity created
+    Entity taskEntity = new Entity("UserComment");
```

Comment:
Renaming this to like commentEntity is the only thing I would change.

Reply:
Thank you sir. I will make the change immeditely.